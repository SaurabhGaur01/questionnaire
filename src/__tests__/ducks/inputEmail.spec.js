import {
    setEmail,
    clearEmail,
    inputEmailReducer,
} from '../../ducks/userInput/inputEmail';

describe('inputEmail Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setEmail should match the snapshot' , () => {
            expect(setEmail(mockObject)).toMatchSnapshot();
        });
        it('clearEmail should match the snapshot' , () => {
            expect(clearEmail()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(inputEmailReducer()).toEqual("");
        });
        it('should return the passed state if called with no action' , () => {
            expect(inputEmailReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(inputEmailReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(inputEmailReducer('', setEmail(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearEmail' , () => {
            expect(inputEmailReducer('', clearEmail())).toEqual("");
        });
    });
});