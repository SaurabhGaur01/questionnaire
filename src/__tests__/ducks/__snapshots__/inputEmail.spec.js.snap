// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputEmail Duck Tests Action Generator clearEmail should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_EMAIL",
}
`;

exports[`inputEmail Duck Tests Action Generator setEmail should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_EMAIL",
  "value": "mockValue",
}
`;
